STYLEGUIDE.md

```diff
@@ -190,7 +190,22 @@ Keeps functions pure.
 
 ---
 
-### Use “be” verbs for boolean types. (e.g. `should`/`is`/`can`/`does`/`has`)
+### Prefix boolean variables with “to be” verbs.
+
+(e.g. `shouldX`, `isY`, `canZ`, `doesX`, `hasY`, `willZ`)
+
+---
+
+### Follow BEM-ish patterns when hard-coding a `data-testid` or `data-lgid`
+
+1. Prefix with “lg”
+2. Words are separated by underscore `_`
+3. Blocks are separated by dash `-`
+4. Name hierarchy should somewhat match the directory structure
+
+e.g. `data-testid="lg-date_picker-calendar_cell"`
```

Comment:
Should this be `lg-date-picker_calendar-cell`? date-picker is the block

Reply:
No, since "date_picker" is one concept

Reply:
Whats the diff between a block and a concept? I know that we are not fully following BEM but from [docs](https://en.bem.info/methodology/quick-start/#element) it looks like it should be `date-picker` because that's the block.

Reply:
BEM uses dashes to separate words within a block/element, and a double underscore to separate blocks/elements/modifiers.

This proposal is to use a single underscore to separate words, and a dash to separate concepts/elements/blocks 

My main gripe with this syntax is that in most text editors a Double click or `Option`+`ArrowKey` presses treat underscores as _one works_ and dashes as a separator. So If I want to replace the `"calendar_cell"` part I can double click it and paste. Or if I want to skip from the end of that string to the previous element I can just press `Option` + `ArrowLeft`. All this is backwards with standard BEM

Reply:
Can you add this bit to the styleguide? `...use a single underscore to separate words, and a dash to separate concepts/elements/blocks`

Reply:
oh nvm its there!